File: chapter1(1-50)/sixth/seven/prime_10001.cpp
#include<iostream>
using namespace std;
int is_prime(int n){
  int i;
  if(n<=1)
    return 0;
  for(i=2;i*i<=n;i++)
    if(n%i==0)
      return 0;
  return 1;

}
int main(){
  int count=0;
  int i=2;
  int a[100005];
  while(count<10001)
  {
    if(is_prime(i)){
      a[count]=i;
      count++;
    }
    i++;
  }
  cout<<a[count-1]<<endl;
  return 0;
}

File: chapter1(1-50)/sixth/seven/prime_10001.rs
fn is_prime(n:i64) ->bool{
    if n<=1
    {
        return false;
    }
    for i in 2..=(n as f64).sqrt() as i64{
        if n%i==0{
            return false;
        }
    }
    true
}//判断质数

fn main(){
    let mut v:Vec<i64>=Vec::new();
    let mut count=0;
    let mut i=2;
    while count<10001{
        if is_prime(i){
            v.push(i);
            count+=1;
        }
        i+=1;
    }
    println!("{}",v[v.len()-1]);
}
